src/YamlValidator/YamlLoader.cs

```diff
@@ -4,26 +4,39 @@
 using System.Collections.ObjectModel;
 
 namespace Microsoft.PowerPlatform.PowerApps.Persistence.YamlValidator;
+
 public class YamlLoader
 {
     public IReadOnlyDictionary<string, string> Load(string filePath, string pathType)
     {
         var deserializedYaml = new Dictionary<string, string>();
+
         if (pathType == Constants.FileTypeName)
         {
             var fileName = Path.GetFileName(filePath);
             var yamlText = Utility.ReadFileData(filePath);
             deserializedYaml.Add(fileName, yamlText);
-            return new ReadOnlyDictionary<string, string>(deserializedYaml);
         }
-
-        // to do: address edge case of .yml files
-        var files = Directory.GetFiles(filePath, $"*{Constants.YamlFileExtension}");
-        foreach (var file in files)
+        else if (pathType == Constants.FolderTypeName)
         {
-            var fileName = Path.GetFileName(file);
-            var yamlText = Utility.ReadFileData(file);
-            deserializedYaml.Add(fileName, yamlText);
+            // TODO: Determine if flag should be required to specify recursive folder search
+            try
+            {
+                foreach (var filename in Directory.EnumerateFiles(filePath, "*" + Constants.YamlFileExtension, SearchOption.AllDirectories))
```

Comment:
Nit: ```Directory.EnumerateFiles(filePath, "*" + Constants.YamlFileExtension, SearchOption.AllDirectories)``` -> move this to a separate line, makes the foreach less verbose and cleaner